open-craft/course-discovery
# BB-4730: Retrieve price from e-commerce response body instead of from stockrecords.

This PR changes the strategy for setting the price when loading data from the e-commerce service: prices are now retrieved using the `price` attribute on the response body, which is calculated using the proper pricing strategy on e-commerce.

**Use case:**
Loading prices on discovery for instances that want to include tax-inclusive prices and display them in the marketing website.

**Testing instructions:**
1. Spin up a **Koa** devstack.
2. Checkout this branch on `course-discovery` and the branch from https://github.com/open-craft/ecommerce/pull/30 in your `ecommerce` repo.
3. From the devstack folder, run `make discovery-shell` and then `./manage.py refresh_course_metadata`.
4. Check that the command completed succesfully.
5. Check that the seat price is 149 here: http://localhost:18381/admin/course_metadata/seat/ (login using http://localhost:18381/login).
5. In the `ecommerce` repo change `ecommerce/settings/base.py`: set `TAX_RATE = '0.5'`.
3. From the devstack folder, run `make discovery-shell` and then `./manage.py refresh_course_metadata`.
4. Check that the seat price is `223.5` (149 + 50% tax).

**Reviewers:**
- [x] @tinumide
